osu.Game.Rulesets.Mania/Objects/Drawables/DrawableHoldNote.cs

```diff
@@ -249,7 +235,7 @@ protected override void Update()
                 //
                 // As per stable, this should not apply for early hits, waiting until the object starts to touch the
                 // judgement area first.
-                if (Head.IsHit && releaseTime == null && DrawHeight > 0)
+                if (Head.IsHit && isHolding.Value && DrawHeight > 0)
```

Comment:
The use of `releaseTime` here is an important distinction, because it's not reset on `endHold()`. Changing this means that the head position is re-adjusted every press, see the following comparison:

Master:

https://github.com/user-attachments/assets/fd8283f6-ea13-449b-9c81-7f1bfc32eb34

This PR:

https://github.com/user-attachments/assets/89423047-eb2f-4f31-8c99-ee0ee35cace3

Reply:
Hmm, right. https://github.com/ppy/osu/pull/33194/commits/bd703b60f90cb661738cdb0a5a5ef3b0d347146b should do it, I think.

Are you ok with no tests for this? It's a bit annoying to exercise this behaviour in tests.

Reply:
That's understandable, imo.